fdbserver/ptxn/test/RealStorageServer.actor.cpp

```diff
@@ -145,28 +154,39 @@ struct StorageServerTestDriver : ServerTestDriver {
 } // namespace ptxn
 
 TEST_CASE("fdbserver/ptxn/test/storageserver") {
-
 	state ptxn::StorageServerTestDriver driver;
 
-	// Set the mock cursor with some input.
+	// TODO: Manage all props in the driver and set default values when the PR is finalized.
+	int nMutationsPerMore = params.getInt("nMutationsPerMore").get();
+	Optional<int> maxMutations = params.getInt("maxMutations").castTo<int>();
+	state int advanceVersionsPerMutation = params.getInt("advanceVersionsPerMutation").get();
+
 	Arena arena;
-	MutationRef mutation(arena, MutationRef::SetValue, "Hello"_sr, "World"_sr);
-	std::string message = BinaryWriter::toValue(mutation, AssumeVersion(currentProtocolVersion)).toString();
-	std::vector<Tag> tags = { driver.tag };
-	state std::vector<MockPeekCursor::OneVersionMessages> allVersionMessages = { MockPeekCursor::OneVersionMessages(
-		8, // version
-		{ MockPeekCursor::MessageAndTags(message, tags) }) };
-	driver.mockPeekCursor = makeReference<MockPeekCursor>(allVersionMessages, arena);
-
-	loop choose{ when(wait(ptxn::StorageServerTestDriver::runStorageServer(&driver))){
-		std::cout << "Storage serves exited unexpectedly" << std::endl;
-	ASSERT(false);
+	VectorRef<Tag> tags;
+	tags.push_back(arena, driver.tag);
+	auto supplier = VersionedMessageSupplier(0, tags, advanceVersionsPerMutation);
+	driver.mockPeekCursor = makeReference<MockPeekCursor>(nMutationsPerMore, maxMutations, supplier, arena);
+
+	int verifyId = params.getInt("verifyId").get();
+	state Future<Void> verify = ptxn::StorageServerTestDriver::verifyGetValueFromId(
+	    &driver,
+	    verifyId,
+	    // Other versions after commitVersion should work too.
+	    VersionedMessageSupplier::commitVersion(verifyId, advanceVersionsPerMutation));
+
+	loop choose{
+		when(wait(ptxn::StorageServerTestDriver::runStorageServer(&driver))) {
+		    std::cout << "Storage serves exited unexpectedly" << std::endl;
+	        ASSERT(false);
         }
-        when(wait(ptxn::StorageServerTestDriver::verifyGetValue(&driver, "Hello"_sr, 8, "World"_sr))) {
+        when(wait(verify)) {
 	        break;
         }
         when(wait(driver.actors.getResult())) {}
     };
 
+    // TODO: Visualize trace log to see how the version in memory and the version in storage advanced. Problem 1: No
+    // trace log sometimes. Problem 2: How to not run as simulation?
```

Comment:
To run as not simulation, pass `-r test` instead of `-r simulation`. This PR should be instructive of how to do so on CI: https://github.com/apple/foundationdb/pull/4983